### PR TITLE
Fix lints and warnings accumulated in dartdoc

### PR DESCRIPTION
**Pull request by jcollins-g**
Most are due to recent library changes, but the rest have just been building up in dartdoc for a long time.  This cleans them up.

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ homepage: https://github.com/dart-lang/dartdoc
 # For development, recommend 1.25.0-dev.0.0 or higher to allow
 # dartanalyzer to work on dartdoc itself.
 environment:
-  sdk: '>=1.23.0-dev.11.5 <2.0.0'
+  sdk: '>=2.0.0-dev.9.0 <3.0.0'
```
Comment (kevmoo) [inline pubspec.yaml]: 👍

Comment (pq) [reply pubspec.yaml]: This constraint is handy to know.  Thanks!

File: testing/test_package/lib/fake.dart [hunk 2]
```diff
@@ -224,6 +221,7 @@ typedef int LotsAndLotsOfParameters(so, many, parameters, it, should, wrap,
 
 /// This class is cool!
 class Cool {
+  // ignore: missing_return
   Cool returnCool() {}
```
Comment (pq) [inline testing/test_package/lib/fake.dart]: Aside: why not just return `null`? In general, I'm curious why all the ignores and not just fixes?

Comment (jcollins-g) [reply testing/test_package/lib/fake.dart]: The test package is intended to be a bit strange, using outdated, error-prone and/or obsolete constructions in some places.   So I don't want to accidentally make the test package too "clean".

I shouldn't be doing that anywhere _outside_ testing/ though.

Comment (pq) [reply testing/test_package/lib/fake.dart]: Makes sense.  Thanks!

**Pull request merged**